notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+
+Notice that $\vec{2x}$ and $\vec{y}$ have the direction but different lengths. They are not equal.
+
+To add two vectors $\vec{y} + \vec{z}$, take one step according to the length of $\vec{y}$, then immediately take one step according to the length of $\vec{z}$ (or vice versa). This is also known as triangle method, where you place the initial point of a vector on the terminal point of the other.
+
+<img src="../../notebooks-images/vector_space_review__adding.png" width="300" />
+
+#### Vector Notations
+Vectors are usually represented as Cartesian Coordinates.
+
+<img src="../../notebooks-images/vector_space_review__notation.png" width="500" />
+
+$$ \vec{x} = \begin{bmatrix} 1 \\ 4  \end{bmatrix} , \quad 
+   \vec{y} = \begin{bmatrix} 3 \\ 2  \end{bmatrix} , \quad 
+   \vec{z} = \begin{bmatrix} 4 \\ 0  \end{bmatrix}$$
+   
+In this notation, arithmetic operations we saw earlier become quite easy.
+
+$$ \vec{2x} = \begin{bmatrix} 2 \\ 8  \end{bmatrix} , \quad
+   \vec{-0.5z} = \begin{bmatrix} -2 \\ 0  \end{bmatrix} , \quad
+   \vec{2x + -0.5z} = \begin{bmatrix} 0 \\ 8  \end{bmatrix}
+$$
+
+<img src="../../notebooks-images/vector_space_review__notation_arithmetic.png" width="500" />
+
+
+Vectors can be added and scaled element-wise:
+
+$$a \vec{x} + b \vec{y} = \begin{bmatrix} a \ x_1 & + & b \  y_1 \\
+                                            & \vdots & \\
+                                          a \  x_n & + & b \  y_n
+                          \end{bmatrix}
+$$
+
+#### The $\vec{1}$ Vector
+In any $d$ dimensional space, $\vec{1}$ is the vector of all $1$'s: 
+$ \begin{bmatrix} 1 \\ \vdots \\ 1  \end{bmatrix}$
+
+#### Span
+The span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$ is the set of all possible linear combinations. For these $p$ vectors:
+
+$$ \{ c_1 \ \vec{v_1} + c_2 \ \vec{v_2} + \dots + c_p \ \vec{v_p} \ : \ \forall c_i \in F\}$$
+
+where $F$ is the field of the vector space (out of scope).
+
+#### Vector Spaces
+A vector space $V$ is the span of a set of vectors $\{\vec{v_1}, \vec{v_2}, \dots, \vec{v_p}\}$, where each $\vec{v_i}$ is a $n \times 1$ dimensional column vector.
+
+#### Vector Subspaces
+A subspace $U$ of $V$ is the span of a set of vectors $\{\vec{u_1}, \dots, \vec{u_q}\}$ where each $\vec{u_i} \in V$. This means every vector in $U$ is also in $V$.
+
+#### Vector Angles
+When you put any two vectors terminal end to terminal end without changing their direction, you can measure the angle between them.
+
+<img src="../../notebooks-images/vector_space_review__angle.png" width="300" />
+
+
+#### Vector Length
+
+Intuition in $\mathbb{R}^2$ :
+
+Recall the triangle method of adding two vectors. If we add two perpendicular vectors $\vec{a} + \vec{b}$ in $\mathbb{R}^2$, then we know that the resulting vector will be the hypotenuse. In this case, we also know that the length of $\vec{a} + \vec{b}$ will follow the Pythagorean Theorem: $\sqrt{a^2 + b^2}$.
```

Comment:
HAHAHHAA, i thought you misspelled "hypotenuse" so i google'd it, but you are correct!
Nice intuition here ^_^